cpp/include/raft/matrix/batched_rearrange.cuh

```diff
@@ -15,73 +15,78 @@
+ * in the scratch space and copied back to the corresponding locations
+ * in the input matrix
+ *
  * @tparam value_idx
  * @tparam value_t
+ *
  * @param[in] handle raft handle
- * @param[out] in input matrix (n_rows * n_cols)
+ * @param[inout] in input matrix (n_rows * n_cols)
  * @param[in] map map containing the order in which rows are to be rearranged (n_rows)
- * @param  D Number of columns of the input/output matrices
- * @param  N Number of rows of the input matrix
- * @param col_batch_size column batch size
+ * @param[in] col_batch_size column batch size
  */
 template <typename T, typename IdxT>
 void batched_gather(raft::device_resources const& handle,
-                    T* in,
-                    IdxT* map,
-                    size_t D,
-                    size_t N,
+                    raft::device_matrix_view<T, IdxT, raft::layout_c_contiguous> in,
+                    raft::device_vector_view<IdxT, IdxT, raft::layout_c_contiguous> map,
                     size_t col_batch_size)
 {
- RAFT_EXPECTS(0 < col_batch_size && col_batch_size <= D, "col_batch_size should be > 0 and <= D");
-  detail::batched_gather(handle,
-                         in,
-                         map,
-                         D,
-                         N,
-                         col_batch_size);
+  IdxT m       = in.extent(0);
+  IdxT n       = in.extent(1);
+  IdxT map_len = map.extent(0);
+  RAFT_EXPECTS(0 < col_batch_size && col_batch_size <= (size_t)n,
+               "col_batch_size should be > 0 and <= n");
+  RAFT_EXPECTS(map_len == m, "size of map should be equal to the number of rows in input matrix");
+
+  detail::batched_gather(handle, in.data_handle(), map.data_handle(), n, m, col_batch_size);
 }
 
 /**
- * In-place scatter elements in a row-major matrix according to a
+ * @brief In-place scatter elements in a row-major matrix according to a
  * map. The length of the map is equal to the number of rows.
  * Batching is done on columns and an additional scratch space of
  * shape n_rows * cols_batch_size is created. For each batch, chunks
  * of columns from each row are copied into the appropriate location
  * in the scratch space and copied back to the corresponding locations
- * in the input matrix. * @tparam value_idx
+ * in the input matrix
+ *
+ * @tparam value_idx
  * @tparam value_t
+ *
  * @param[in] handle raft handle
- * @param[out] in input matrix (n_rows * n_cols)
- * @param[in] map map containing the destination index for each row (n_rows)
- * @param  D Number of columns of the input/output matrices
- * @param  N Number of rows of the input matrix
- * @param col_batch_size column batch size
+ * @param[inout] in input matrix (n_rows * n_cols)
+ * @param[in] map map containing destination index of each row (n_rows)
+ * @param[in] col_batch_size column batch size
  */
 template <typename T, typename IdxT>
 void batched_scatter(raft::device_resources const& handle,
```

Comment:
We should rename this function to `scatter_inplace`. That explain what it is doing. The 'batching` is an implementation detail imho.

Reply:
This is a deleted function. `batched_rearrange.cuh` is deleted.